### ISSUE TITLE
Invalid build command due to semicolon in compile flags

### ISSUE THREAD

**Issue by RT-Bow** [PointCloudLibrary/pcl#4865]
**Bug description**

When building PCL from source using cmake, flags can be set using the `SSE_FLAGS` variable. This variable contains a list of flags and is currently used as a string and appended to `CMAKE_CXX_FLAGS`. This can then lead to a misconfiguration of the build commands since the string conversion will keep the semicolon used as a separator in the original list, causing the command to exit early.

**Context**

I have spotted this issue when trying to build PCL 1.12 release from source and also building PCL dependencies from source, without the use of a packet manager.

**Expected behavior**

When generating the many `.make` files using cmake, no semicolon should be present in the `CXX_FLAGS` variable set in any of the `*/flags.make` files.

**Current behavior**

As an example, the generated `kdtree/CMakeFiles/pcl_kdtree.dir/flags.make` might contain something like that:
```
CXX_FLAGS =  -Wabi=11 -Wall -Wextra -Wno-unknown-pragmas -fno-strict-aliasing -Wno-format-extra-args -Wno-sign-compare -Wno-invalid-offsetof -Wno-conversion -march=native;-ffloat-store -O3 -DNDEBUG -fPIC   -std=c++14
```

**To Reproduce**

Apologies for leaving this one blank but I do not have the means/knowledge to test which part of my configuration leads to this behavior.

**Environment**

 - OS: Ubuntu 18.04
 - Compiler: GCC 9.4.0
 - PCL Version: 1.12.0

**Possible Solution**

I believe the offending part of the code comes from the main `CMakeLists.txt` file (line 107 on master at this time):
```cmake
  if("${CMAKE_CXX_FLAGS}" STREQUAL "${CMAKE_CXX_FLAGS_DEFAULT}")
    if (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 7)
      string(APPEND CMAKE_CXX_FLAGS " -Wabi=11")
    else()
      string(APPEND CMAKE_CXX_FLAGS " -Wabi")
    endif()
    string(APPEND CMAKE_CXX_FLAGS " -Wall -Wextra -Wno-unknown-pragmas -fno-strict-aliasing -Wno-format-extra-args -Wno-sign-compare -Wno-invalid-offsetof -Wno-conversion ${SSE_FLAGS}")
  endif()
```
I have fixed it locally by the following:
```cmake
  if("${CMAKE_CXX_FLAGS}" STREQUAL "${CMAKE_CXX_FLAGS_DEFAULT}")
    if (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 7)
      string(APPEND CMAKE_CXX_FLAGS " -Wabi=11")
    else()
      string(APPEND CMAKE_CXX_FLAGS " -Wabi")
    endif()
    string(APPEND CMAKE_CXX_FLAGS " -Wall -Wextra -Wno-unknown-pragmas -fno-strict-aliasing -Wno-format-extra-args -Wno-sign-compare -Wno-invalid-offsetof -Wno-conversion")
    foreach(FLAG IN LISTS SSE_FLAGS)
      string(APPEND CMAKE_CXX_FLAGS " ${FLAG}")
    endforeach()
  endif()
```
I am sure there will be a more elegant fix, if any needed, hence this is not on pull request.

**Misc**

This is my first time contributing to PCL (or anything on github for that matter) so advice on good practice are welcome.

**Comment by mvieth** [comment]
There is a pull request in the works (#4698) which might fix this problem. It would be great if you could have a look at that and check whether anything needs to be changed.

> This is my first time contributing to PCL (or anything on github for that matter) so advice on good practice are welcome.

Your bug report is good! Very detailed and even a potential fix :+1:

**Comment by RT-Bow** [comment]
Thanks, glad to help.

> There is a pull request in the works (#4698) which might fix this problem. It would be great if you could have a look at that and check whether anything needs to be changed.


I had a look and you're correct, [this](https://github.com/PointCloudLibrary/pcl/pull/4698/commits/e25b64ed0c68b3cbfe422e193b5e517eb98afd28) recent commit will fix the issue. I'll resolve the issue when the PR gets through.

**Comment by kunaltyagi** [comment]
Closed by #4698
